codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/RestJson.kt

```diff
@@ -56,15 +62,32 @@ class RestJsonFactory : ProtocolGeneratorFactory<HttpBoundProtocolGenerator> {
  * `application/json` if not overridden.
  */
 class RestJsonHttpBindingResolver(
-    model: Model,
+    private val model: Model,
     contentTypes: ProtocolContentTypes,
 ) : HttpTraitHttpBindingResolver(model, contentTypes) {
     /**
      * In the RestJson1 protocol, HTTP responses have a default `Content-Type: application/json` header if it is not
      * overridden by a specific mechanism e.g. an output shape member is targeted with `httpPayload` or `mediaType` traits.
      */
-    override fun responseContentType(operationShape: OperationShape): String =
-        super.responseContentType(operationShape) ?: "application/json"
+    override fun responseContentType(operationShape: OperationShape): String? {
+        val members = operationShape
+            .outputShape(model)
+            .members()
+        // temporary fix for https://github.com/awslabs/smithy/blob/df456a514f72f4e35f0fb07c7e26006ff03b2071/smithy-model/src/main/java/software/amazon/smithy/model/knowledge/HttpBindingIndex.java#L352
+        // https://github.com/awslabs/smithy/issues/1259
```

Comment:
```suggestion
        // Temporary fix for https://github.com/awslabs/smithy/blob/df456a514f72f4e35f0fb07c7e26006ff03b2071/smithy-model/src/main/java/software/amazon/smithy/model/knowledge/HttpBindingIndex.java#L352
        // https://github.com/awslabs/smithy/issues/1259
```

TODOs let us grep through the codebase later.